### PR TITLE
e2e network policy enclose IPv6 destinations

### PR DESCRIPTION
**Pull request by aojea**
We moved from testing against a DNS destination to use IP:Port, 
so we need to be sure that IPv6 addresses are enclosed in square 
brackets with that format.

/kind bug

The test now fails with:

> Nov 25 08:14:46.564: INFO: stdout: "UNKNOWN: address fd00:10:96::8198:80: too many colons in address\nUNKNOWN: address fd00:10:96::8198:80: too many colons in address\nUNKNOWN: address fd00:10:96::8198:80: too many colons in address\nUNKNOWN: address fd00:10:96::8198:80: too many colons in address\nUNKNOWN: address fd00:10:96::8198:80: too many colons in address\n"
> Nov 25 08:14:46.564: INFO: 
> Last 100 log lines of client-can-connect-80-r9gjz:
> UNKNOWN: address fd00:10:96::8198:80: too many colons in address
> UNKNOWN: address fd00:10:96::8198:80: too many colons in address
> UNKNOWN: address fd00:10:96::8198:80: too many colons in address
> UNKNOWN: address fd00:10:96::8198:80: too many colons in address
> UNKNOWN: address fd00:10:96::8198:80: too many colons in address

because of

>     Command:
>       /bin/sh
>     Args:
>       -c
>       for i in $(seq 1 5); do /agnhost connect fd00:10:96::8198:80 --protocol tcp --timeout 8s && exit 0 || sleep 1; done; exit 1
>     State:          Terminated
> 

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/network_policy.go [hunk 1]
```diff
@@ -2200,10 +2198,12 @@ var _ = SIGDescribe("NetworkPolicy API", func() {
 		npVersion := "v1"
 		npClient := f.ClientSet.NetworkingV1().NetworkPolicies(ns)
 		npTemplate := &networkingv1.NetworkPolicy{
-			ObjectMeta: metav1.ObjectMeta{GenerateName: "e2e-example-netpol",
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "e2e-example-netpol",
 				Labels: map[string]string{
 					"special-label": f.UniqueName,
-				}},
+				},
```
Comment (aojea) [inline test/e2e/network/network_policy.go:2205]: this is my linter things :shrug:

**Comment by spiffxp** [comment]
/milestone v1.20
/hold
@kubernetes/release-team I think this should be included in v1.20 vs. 1.20.1.  Please remove /hold if you agree, or remove from milestone if you disagree.
- pro: fixes a network policy test when ipv6 is used, we landed dual stack services in alpha as part of v1.20
- con: tests aren't release-blocking, the enhancement is only alpha, this will change the SHA if we're looking for continuous passes on a SHA

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by saschagrunert** [comment]
Hey @spiffxp and @aojea, we're about to cut v1.20.0-rc.0 which also creates the release-1.20 branch. From that point on we have to cherry-pick all changes which should be included in v1.20.0.

From my perspective this PR is fine to be merged, but I'd like to give @jeremyrickard a head nod.

**Comment by aojea** [comment]
thanks @saschagrunert , @jeremyrickard I appreciate if we can have it in 1.20, as you can see, network policies tests with ipv6  doesn't work :shrug:

**Review by hasheddan — COMMENTED** [review]
@aojea @spiffxp thanks for the fix and context here! While the blast radius on this change is rather small, the fact that this is a non-blocking test-only fix indicates that we should not break code / test freeze for inclusion. Would love to see this cherry-picked for v1.20.1 :)

/milestone clear

**Comment by aojea** [comment]
/hold cancel
master reopened, we should be good to go

**Comment by aojea** [comment]
/retest

This sounds familiar :)
Kubernetes e2e suite: [sig-api-machinery] ResourceQuota should create a ResourceQuota and capture the life of a pod. [Conformance] expand_less

**Pull request merged**